analysis/estimate/notebooks/run-estimate.Rmd

```diff
@@ -0,0 +1,317 @@
+  tidyr::pivot_wider(
+    names_from = expression_type, 
+    values_from = score
+  )
+
+ggplot(expr_wide_df) + 
+  aes(x = bulk_counts, y = bulk_tpm) + 
+  geom_point() +
+  geom_abline(color = "darkorange2") +
+  coord_equal() +
+  facet_grid(
+    rows = vars(score_type), 
+    cols = vars(project_id)
+  ) 
+```
+
+Across all projects, we see that immune scores tend to be slightly higher for TPM, and by contrast stromal scores tend to be slightly lower for TPM compared to bulk.
+The effect sizes are fairly small, but the effects seem likely to be significant, which we'll check separately for `stromal` and `immune`.
+Below, we fit regressions that fix the slope to 1, so the intercept will tell us the bias from `y=x`:
+
+```{r}
+stromal_df <- expr_wide_df |>
+  dplyr::filter(score_type == "stromal")
+summary(lm(bulk_tpm ~ 1 + offset(bulk_counts), data = stromal_df))
+
+
+immune_df <- expr_wide_df |>
+  dplyr::filter(score_type == "immune")
+summary(lm(bulk_tpm ~ 1 + offset(bulk_counts), data = immune_df))
+
+```
+
+Indeed, effect sizes are not very large in the scheme of the full range of values, but they are highly significant: Compared to bulk counts, immune scores are relatively higher when calculated with TPM, and stromal scores are relatively lower when calculated with TPM.
+
+### Compare measures from pseudobulk and bulk
+
+Since bulk scores were very similar, we'll proceed to visualize only the bulk counts.
+
+```{r}
+estimate_sub_df <- estimate_df |>
+  dplyr::filter(expression_type != "bulk_tpm")
+```
+
+First, we'll look at both the distribution of scores, and then the relationship between pseudobulk and bulk scores.
+
+#### Score distributions
+
+```{r fig.height=4, fig.width=12}
+ggplot(estimate_sub_df) + 
+  aes(x = score_type, y = score, color = expression_type) + 
+  geom_jitter(
+    position = position_jitterdodge(jitter.height = 0, jitter.width = 0.1), 
+    size = 0.75
+  ) +
+  scale_color_brewer(palette = "Set2") +
+  facet_wrap(vars(project_id), scales = "free_x", nrow = 1) + 
+  theme(legend.position = "bottom") + 
+  guides(color = guide_legend(override.aes = list(size = 2)))
+```
+
+First, what do we see for immune scores?
+
+* For SCPCP000001 and SCPCP000002, pseudobulk scores tend to be higher than bulk scores
+  * However, for SCPCP000001, we have a sort of trimodal distribution of values - it would be nice if this corresponded to diagnoses?!
+* For the other three projects, pseudobulk scores appear to be somewhat lower than bulk
+
+Second, what do we see for stromal scores?
+
+* For SCPCP000001 and SCPCP000002, pseudobulk scores look either on par with bulk or _maybe_ a bit lower, but that is mostly in SCPCP000001 and one outlier in SCPCP000002
+* For the other three projects, pseudobulk and bulk scores appear highly similar
+
+
+#### Compare modalities
+
+```{r fig.width=10}
+compare_modalities_df <- estimate_df |>
+  tidyr::pivot_wider(
+    names_from = expression_type, 
+    values_from = score
+  ) 
```

Comment:
I think you do this twice and can remove this?

Reply:
well it sure sounds like something I'd do 😂